### PR TITLE
recipe for pyqtwebkit

### PR DESCRIPTION
**Pull request by gillins**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by gillins** [comment]
Merge of `pyqt` and https://github.com/osgeo-forge/pyqtwebkit-feedstock recipes. Not sure if this is the best approach?

ping: @SrNetoChan @dakcarto  

xref: https://github.com/conda-forge/qgis-feedstock/issues/68

**Comment by dakcarto** [comment]
> Merge of `pyqt` and https://github.com/osgeo-forge/pyqtwebkit-feedstock recipes. Not sure if this is the best approach?

This is what I originally did for pyqtwebkit-feedstock 5.9.{2,7}, based it upon the `pyqt` recipe. Looks good here, though I have not built it.

File: recipes/pyqtwebkit/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  license_file: LICENSE
+  summary: 'Python binding of the cross-platform GUI toolkit Qt. QtWebKit only.'
+  description: |
+    "PyQt is a set of Python v2 and v3 bindings for The Qt Company's Qt
+    application framework and runs on all platforms supported by Qt including
+    Windows, MacOS/X and Linux. PyQt5 supports Qt v5. PyQt4 supports Qt v4 and
+    will build against Qt v5. The bindings are implemented as a set of Python
+    modules and contain over 1,000 classes."
+  doc_url: https://www.riverbankcomputing.com/software/pyqt/
+  dev_url: https://github.com/pyqt
```
Comment (dakcarto) [inline recipes/pyqtwebkit/meta.yaml:91]: This seems to be an unofficial `pyqt` dev URL.

**Comment by gillins** [comment]
Anyone know why the Linux builds fail? Something to do with opengl? Works fine locally for me with an Xsession...

@ccordoba12 @mingwandroid @ocefpaf @dakcarto @SrNetoChan are you all happy to be maintainers on this?

**Comment by dakcarto** [comment]
@gillins wrote:
> Anyone know why the Linux builds fail? Something to do with opengl? Works fine locally for me with an Xsession...

The QPainter seems to be unable to initialize. Segfault probably follows after a call on that null pointer. A GUI session will be required to test rendering of a web page. Probably requires a dive into the GUI session capabilities of the Linux build container. Maybe there are other examples of recipes that test GUI functions?

While not a fix, you might try a simpler test URL (https://example.com) and redirect (https://example.org). Google's standard landing pages are not exactly simple under the hood. In this way, at least, graphics are not attempted to be rendered.

**Comment by SrNetoChan** [comment]
@gillins thanks for your work. Under my humble capacities, I am happy to help maintain this recipe.

**Comment by mingwandroid** [comment]
I don't have much bandwidth to help with this one, sorry.

**Comment by gillins** [comment]
Thanks @dakcarto  but didn't seem to make any difference. I've commented this test out on Linux for now - perhaps we can revisit later. Definitely works for me locally.

I think this one is ready to go...

**Comment by SrNetoChan** [comment]
I say let's merge it and create an issue to remember the commented test.

This way I can include this package on the qgis recipe. And we can test if the failing test affect other machines.

**Comment by SrNetoChan** [comment]
Not merge... Remove from stagging.

**Comment by ocefpaf** [comment]
> Merge of `pyqt` and [osgeo-forge/pyqtwebkit-feedstock](https://github.com/osgeo-forge/pyqtwebkit-feedstock) recipes. Not sure if this is the best approach?

It is a fine approach for testing when you don't have access to local machines with all platforms. We cannot merge existing recipes though. I see you only have `pyqtwebkit` and it is working, merge?

**Comment by gillins** [comment]
Sorry I didn't word that very clearly... This is a new recipe for `conda-forge`. I took the recipe from `osgeo-forge` and updated with the contents of our `pyqt` so it works against our current `qt`. 

It is *only* the Python bindings for `qtwebkit` that get built, but there is a fair bit of duplication with `pyqt`. I thought on balance it was better to have it separate from `pyqt` as that recipe is starting to get quite complex plus most people using `pyqt` don't need `qtwebkit` so enforcing them to install that also would be unnecessary.

Is this ok?

**Comment by ocefpaf** [comment]
> Sorry I didn't word that very clearly... This is a new recipe for `conda-forge`. I took the recipe from `osgeo-forge` and updated with the contents of our `pyqt` so it works against our current `qt`.

Sorry. I misunderstood!

> It is _only_ the Python bindings for `qtwebkit` that get built, but there is a fair bit of duplication with `pyqt`. I thought on balance it was better to have it separate from `pyqt` as that recipe is starting to get quite complex plus most people using `pyqt` don't need `qtwebkit` so enforcing them to install that also would be unnecessary.
> 
> Is this ok?

Yep. Makes perfect sense and it is easier to maintain. Let's merge this!

**Pull request merged**